azure/services/virtualmachines/virtualmachines_test.go

```diff
@@ -283,3 +291,78 @@ func TestDeleteVM(t *testing.T) {
+			expect: func(s *mock_virtualmachines.MockVMScopeMockRecorder, i *mock_identities.MockClientMockRecorder) {
+				i.GetClientID(gomockinternal.AContext(), fakeUserAssignedIdentity.ProviderID).AnyTimes().Return(fakeUserAssignedIdentity.ProviderID, nil)
+			},
+		},
+		{
+			name:             "matching user assigned identities",
+			specIdentities:   []infrav1.UserAssignedIdentity{fakeUserAssignedIdentity},
+			actualIdentities: []infrav1.UserAssignedIdentity{fakeUserAssignedIdentity},
+			expect: func(s *mock_virtualmachines.MockVMScopeMockRecorder, i *mock_identities.MockClientMockRecorder) {
+				i.GetClientID(gomockinternal.AContext(), fakeUserAssignedIdentity.ProviderID).AnyTimes().Return(fakeUserAssignedIdentity.ProviderID, nil)
+			},
+		},
+		{
+			name:             "less user assigned identities than expected",
+			specIdentities:   []infrav1.UserAssignedIdentity{fakeUserAssignedIdentity, fakeUserAssignedIdentity2},
+			actualIdentities: []infrav1.UserAssignedIdentity{fakeUserAssignedIdentity},
+			expect: func(s *mock_virtualmachines.MockVMScopeMockRecorder, i *mock_identities.MockClientMockRecorder) {
+				i.GetClientID(gomockinternal.AContext(), fakeUserAssignedIdentity.ProviderID).AnyTimes().Return(fakeUserAssignedIdentity.ProviderID, nil)
+				i.GetClientID(gomockinternal.AContext(), fakeUserAssignedIdentity2.ProviderID).AnyTimes().Return(fakeUserAssignedIdentity2.ProviderID, nil)
+				s.SetConditionFalse(infrav1.VMIdentitiesReadyCondition, infrav1.UserAssignedIdentityMissingReason, clusterv1.ConditionSeverityWarning, "VM is missing expected user assigned identity with client ID: "+fakeUserAssignedIdentity2.ProviderID).Times(1)
+			},
+		},
+		{
+			name:             "more user assigned identities than expected",
+			specIdentities:   []infrav1.UserAssignedIdentity{fakeUserAssignedIdentity},
+			actualIdentities: []infrav1.UserAssignedIdentity{fakeUserAssignedIdentity, fakeUserAssignedIdentity2},
+			expect: func(s *mock_virtualmachines.MockVMScopeMockRecorder, i *mock_identities.MockClientMockRecorder) {
+				i.GetClientID(gomockinternal.AContext(), fakeUserAssignedIdentity.ProviderID).AnyTimes().Return(fakeUserAssignedIdentity.ProviderID, nil)
+			},
+		},
+		{
+			name:             "mismatched user assigned identities by content",
+			specIdentities:   []infrav1.UserAssignedIdentity{fakeUserAssignedIdentity},
+			actualIdentities: []infrav1.UserAssignedIdentity{fakeUserAssignedIdentity2},
+			expect: func(s *mock_virtualmachines.MockVMScopeMockRecorder, i *mock_identities.MockClientMockRecorder) {
+				i.GetClientID(gomockinternal.AContext(), fakeUserAssignedIdentity.ProviderID).AnyTimes().Return(fakeUserAssignedIdentity.ProviderID, nil)
+				s.SetConditionFalse(infrav1.VMIdentitiesReadyCondition, infrav1.UserAssignedIdentityMissingReason, clusterv1.ConditionSeverityWarning, "VM is missing expected user assigned identity with client ID: "+fakeUserAssignedIdentity.ProviderID).Times(1)
+			},
+		},
+	}
```

Comment:
If the same identity could be listed more than once in either the VM or CAPZ spec, then adding some test cases like that would be worthwhile.

Reply:
I definitely can't add the same identity twice on the portal, but I'll try specifying the same one in the spec and see if this is something we need to test.

Reply:
Looks like specifying duplicate identities in the spec just results in one identity being listed on the VM. This won't cause any errors with the current implementation, but I'll go ahead and add a test case for it since it's possible to list them on the spec.